Rework developer landing page for topic-oriented discovery (#517)
canonical/launchpad-manual: workfile.md

- [Navigation menus](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/navigation-menus.rst)
- [URL traversal](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/url-traversal.rst)
- [Registry](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/registry.rst)
- [Engineering overview translations](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/engineering-overview-translations.rst)
- [Application framework](docs/developer/explanation/core-components-and-set-up/framework.rst)
- [Code concepts](docs/developer/explanation/core-components-and-set-up/concepts.rst)
- [The Launchpad PPA](docs/developer/explanation/core-components-and-set-up/launchpad-ppa.rst)
- [Navigating the tree](docs/developer/explanation/core-components-and-set-up/navigating.rst)
- [Launchpad pip integration](docs/developer/explanation/core-components-and-set-up/pip.rst)
- [Launchpad installation details](docs/developer/explanation/core-components-and-set-up/running-details.rst)
- [Building live filesystems](docs/developer/explanation/core-components-and-set-up/building-live-filesystems.rst)

> - [Preserve query count](docs/developer/how-to/launchpad-development-tips/preserve-query-count.rst)

Reviewer: This doesn't seem like a core concept

Author: Thanks, this slipped into the wrong section.

## Database
- [Database overview](docs/developer/explanation/database/index.rst)
- [Database performance](docs/developer/explanation/database/database-performance.rst)
- [Live database patching](docs/developer/explanation/database/live-patching.rst)
- [PostgreSQL and Launchpad](docs/developer/explanation/database/postgresql.rst)
- [Storm migration guide](docs/developer/explanation/database/storm-migration-guide.rst)
- [Working with db-devel](docs/developer/explanation/database/working-with-db-devel.rst)
- [Rename a database table](docs/developer/how-to/launchpad-development-tips/rename-database-table.rst)

> ## Best Practices

Reviewer: I think we should relabel this as "Development best practices". Best practices isn't too clear

Author: Agree. Thanks!

- [Launchpad permissions](docs/developer/explanation/development-best-practices/security-policy.rst)
- [Template reuse](docs/developer/explanation/development-best-practices/template-reuse.rst)
- [Python style guide](docs/developer/reference/python.rst)
- [XXX policy](docs/developer/explanation/developing-for-launchpad/xxx-policy.rst)

## Frontend development
- [JavaScript build system](docs/developer/explanation/javascript/javascript-buildsystem.rst)
- [Integration testing in JavaScript](docs/developer/explanation/javascript/javascript-integration-testing.rst)
- [Developing with YUI Test](docs/developer/explanation/javascript/javascript-unittesting.rst)
- [Static assets overview](docs/developer/explanation/static-assets/index.rst)
- [CSS](docs/developer/explanation/static-assets/css.rst)
- [CSS sprites](docs/developer/explanation/static-assets/css-sprites.rst)
- [Favicons](docs/developer/explanation/static-assets/favicon.rst)
- [Images](docs/developer/explanation/static-assets/images.rst)
- [Use MockIo library](docs/developer/how-to/launchpad-development-tips/mockio.rst)
- [CSS style guide](docs/developer/reference/css.rst)

> ## Services Reference

Reviewer: We should generalize this title since these are not all reference docs in our unique documentation sense. Our use of a word like reference should be standardized in the company.

Author: Takes notes... the word "reference" is burnt forever :-)

- [Build farm](docs/developer/reference/services/build-farm.rst)
- [Signing service](docs/developer/reference/services/signing.rst)
- [Fetch service](docs/developer/reference/services/fetch-service.rst)
- [Buildbot](docs/developer/reference/services/buildbot.rst)
- [Automatic translations tarball exports](docs/developer/reference/services/automatic-translations-export.rst)
- [Mirror prober](docs/developer/reference/services/mirror-prober.rst)
- [Ubuntu mirrors index](docs/developer/reference/services/ubuntu-mirrors-index.rst)
- [Git hosting](docs/developer/reference/services/git-hosting.rst)
- [Code import](docs/developer/reference/services/code-import.rst)
- [Launchpad public mailing lists archives](docs/developer/reference/services/mailing-lists-archives.rst)
- [Launchpad and email](docs/developer/reference/email.rst)
- [Ubuntu package publishing](docs/developer/explanation/core-components-and-set-up/publisher/ubuntu-package-publishing.rst)
- [Ubuntu archive publisher](docs/developer/explanation/core-components-and-set-up/publisher/ubuntu-archive-publisher.rst)
- [Code](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/code.rst)

> - [Malone](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/malone.rst)

Reviewer: I think this doc could be under core concepts and architecture

Author: Maybe borderline, but I still want to keep this in the services / components section - and this will be a very clear choice once we have reworked that article.

- [Contribute changes](docs/developer/how-to/common-development-tasks/contributing-changes.rst)
- [Use an updated dependency](docs/developer/how-to/common-development-tasks/use-updated-dependency.rst)
- [Database schema changes process](docs/developer/how-to/common-development-tasks/database-schema-changes-process.rst)
- [Apply database schema changes](docs/developer/how-to/common-development-tasks/apply-schema-changes.rst)
- [Creating a new page in Launchpad](docs/developer/tutorials/creating-a-page-in-launchpad.rst)

## Operating Launchpad
- [Get overview of build farm](docs/developer/how-to/operate-launchpad/getting-overview-of-build-farm.rst)
- [Check availability of frontends](docs/developer/how-to/operate-launchpad/checking-availability-of-launchpad-frontends.rst)
- [Manage users and teams in development environments](docs/developer/how-to/operate-launchpad/manage-users.rst)
- [Create a job to publish an artifact](docs/developer/how-to/operate-launchpad/create-job-publish-artifact.rst)
- [Triage Launchpad project bugs](docs/developer/how-to/operate-launchpad/triage-bugs.rst)
- [Tagging bugs about Launchpad](docs/developer/reference/bug-tags.rst)

> - [Land updates for Loggerhead](docs/developer/how-to/operate-launchpad/land-update-for-loggerhead.rst)

Reviewer: I think this doc is no longer useful since Bazaar has been depracated

Author: That is partly true. While we have deprecated Bazaar, we still have Loggerhead deployed, and might need to deploy new versions, so for now we should keep that document. But we can argue whether it needs to be on the landing page. On a second thought... this is most likely rarely used, so we can remove it from the landing page. Thanks!

# Developer docs discovery map

## Goal
Keep the Diataxis structure for the documentation itself, but add a topic-oriented discovery layer for everything under docs/developer so people can find pages by subject instead of only by intent.

This map applies only to docs/developer. The user documentation stays unchanged.

## Setup and first steps

> Get started with Launchpad core development by setting up your local environment and getting the source code. These resources guide you through downloading, installing, and running Launchpad locally.

Reviewer: ```suggestion
Get started with Launchpad core development by downloading the source code, running Launchpad locally, or even deploying Soyuz on your machine for local builds.
```
The main idea behind this edit is something that DP mentioned about the landing page work we did last time, i.e., we should avoid "documenting the documentation". So basically, this paragraph doesn't need to reference itself i.e., "These resources", "in this doc", etc.

Author: Thanks! Makes a lot of sense.

- [Get the Launchpad source code](docs/developer/how-to/get-started/get-source-code.rst)
- [Run Launchpad locally with quick setup](docs/developer/how-to/get-started/running-quickstart.rst)
- [Run Launchpad locally with advanced setup](docs/developer/how-to/get-started/running.rst)
- [Database setup](docs/developer/how-to/get-started/database-setup.rst)
- [Deploy Soyuz locally](docs/developer/how-to/get-started/use-soyuz-locally.rst)
- [Develop with Buildd](docs/developer/how-to/get-started/develop-with-buildd.rst)
- [Use codehosting locally](docs/developer/how-to/launchpad-development-tips/codehosting-locally.rst)

## Day-to-day development

> Find guidelines and instructions for your routine development tasks, from fixing bugs to submitting database schema changes. This section covers the common workflows and processes required to implement and ship changes in Launchpad.

Reviewer: ```suggestion
Guidelines and instructions for routine development tasks, from fixing bugs to submitting database schema changes. Explore common workflows and processes required to implement and ship changes in Launchpad.
```

- [Database schema changes process](docs/developer/how-to/common-development-tasks/database-schema-changes-process.rst)
- [Apply database schema changes](docs/developer/how-to/common-development-tasks/apply-schema-changes.rst)
- [Creating a new page in Launchpad](docs/developer/tutorials/creating-a-page-in-launchpad.rst)
- [Handle exceptions](docs/developer/how-to/launchpad-development-tips/exceptions.rst)
- [Handle security policies](docs/developer/how-to/launchpad-development-tips/security.rst)
- [About Launchpad branches](docs/developer/explanation/developing-for-launchpad/branches.rst)
- [Datetime usage guide](docs/developer/explanation/developing-for-launchpad/datetime-usage.rst)
- [Error explanations](docs/developer/explanation/developing-for-launchpad/error-explanations.rst)
- [Feature flags](docs/developer/explanation/developing-for-launchpad/feature-flags.rst)
- [Journey of a change to production](docs/developer/explanation/developing-for-launchpad/journey-of-a-change-to-production.rst)
- [Pre merge reviews](docs/developer/explanation/developing-for-launchpad/pre-merge-reviews.rst)

## Operating Launchpad

> Learn how to manage, maintain, and monitor Launchpad services and environments in real time. This section covers operating the build farm, checking frontend availability, user management, and utilizing diagnostic CLI tools.

Reviewer: ```suggestion
Manage, maintain, and monitor Launchpad services and environments in real time, including operating the build farm, checking frontend availability, user management, and utilizing diagnostic CLI tools.
```

- [Get overview of build farm](docs/developer/how-to/operate-launchpad/getting-overview-of-build-farm.rst)
- [Check availability of frontends](docs/developer/how-to/operate-launchpad/checking-availability-of-launchpad-frontends.rst)
- [Manage users and teams in development environments](docs/developer/how-to/operate-launchpad/manage-users.rst)
- [Create a job to publish an artifact](docs/developer/how-to/operate-launchpad/create-job-publish-artifact.rst)
- [Triage Launchpad project bugs](docs/developer/how-to/operate-launchpad/triage-bugs.rst)
- [Tagging bugs about Launchpad](docs/developer/reference/bug-tags.rst)
- [Build and publish Launchpad development LXD images](docs/developer/how-to/operate-launchpad/building-lpdev-images.rst)
- [Use lp-shell](docs/developer/how-to/operate-launchpad/use-lp-shell.rst)
- [Generate Launchpad API docs](docs/developer/how-to/launchpad-development-tips/launchpad-api-docs.rst)

## Testing, debugging And troubleshooting

> Ensure the quality of your code and resolve issues effectively using Launchpad's testing and debugging suites. These resources show you how to execute tests, troubleshoot breakpoints, and integrate debugging tools like VS Code.

Reviewer: ```suggestion
Verify the quality of your code and resolve issues effectively using Launchpad's testing and debugging suites. Learn how to execute tests, troubleshoot breakpoints, and integrate debugging tools into VS Code.
```
VS Code isn't the actual debugging tool I think.

- [Navigation menus](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/navigation-menus.rst)
- [URL traversal](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/url-traversal.rst)
- [Registry](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/registry.rst)
- [Engineering overview translations](docs/developer/explanation/core-components-and-set-up/parts-of-launchpad/engineering-overview-translations.rst)
- [Application framework](docs/developer/explanation/core-components-and-set-up/framework.rst)
- [Code concepts](docs/developer/explanation/core-components-and-set-up/concepts.rst)
- [The Launchpad PPA](docs/developer/explanation/core-components-and-set-up/launchpad-ppa.rst)
- [Navigating the tree](docs/developer/explanation/core-components-and-set-up/navigating.rst)
- [Launchpad pip integration](docs/developer/explanation/core-components-and-set-up/pip.rst)
- [Launchpad installation details](docs/developer/explanation/core-components-and-set-up/running-details.rst)
- [Building live filesystems](docs/developer/explanation/core-components-and-set-up/building-live-filesystems.rst)

## Database

> Explore Launchpad's relational database layer, its performance optimization strategies, and schema management practices. This guide provides workflows for executing database migrations, renaming tables, and maintaining query efficiency.

Reviewer: ```suggestion
Explore Launchpad's relational database layer, its performance optimization strategies, and schema management practices. Understand workflows for executing database migrations, renaming tables, and maintaining query efficiency.
```

- [Database overview](docs/developer/explanation/database/index.rst)
- [Database performance](docs/developer/explanation/database/database-performance.rst)
- [Live database patching](docs/developer/explanation/database/live-patching.rst)
- [PostgreSQL and Launchpad](docs/developer/explanation/database/postgresql.rst)
- [Storm migration guide](docs/developer/explanation/database/storm-migration-guide.rst)
- [Working with db-devel](docs/developer/explanation/database/working-with-db-devel.rst)
- [Rename a database table](docs/developer/how-to/launchpad-development-tips/rename-database-table.rst)
- [Preserve query count](docs/developer/how-to/launchpad-development-tips/preserve-query-count.rst)

## Development best practices

> Adhere to established coding style guides, architectural rules, and security policies to maintain the quality of the codebase. These documents outline recommended practices for template reuse, Python styling, and exception handling logic.

Reviewer: ```suggestion
Coding style guides, architectural rules, and security policies used to maintain the quality of the codebase, from use of assertions to preferred Python conventions.
```

- [Architectural guide](docs/developer/explanation/development-best-practices/architecture.rst)
- [Assertions in Launchpad](docs/developer/explanation/development-best-practices/assertions-in-launchpad.rst)
- [Bug triage process background](docs/developer/explanation/development-best-practices/bug-triage-process-background.rst)
- [Charm development](docs/developer/explanation/development-best-practices/charms.rst)
- [Code import in depth](docs/developer/explanation/development-best-practices/codeimport.rst)
- [Page tests](docs/developer/explanation/development-best-practices/page-tests.rst)
- [About Launchpad performance](docs/developer/explanation/development-best-practices/performance.rst)
- [Launchpad permissions](docs/developer/explanation/development-best-practices/security-policy.rst)
- [Template reuse](docs/developer/explanation/development-best-practices/template-reuse.rst)
- [Python style guide](docs/developer/reference/python.rst)
- [XXX policy](docs/developer/explanation/developing-for-launchpad/xxx-policy.rst)

## Frontend development

> Build and test Launchpad's client-side user interface using standard assets and frameworks. This section describes the JavaScript build system, styling standards, and testing interfaces.

Reviewer: ```suggestion
Build and test Launchpad's client-side user interface using standard assets and frameworks. Understand the JavaScript build system, styling standards, and testing interfaces.
```

- [JavaScript build system](docs/developer/explanation/javascript/javascript-buildsystem.rst)
- [Integration testing in JavaScript](docs/developer/explanation/javascript/javascript-integration-testing.rst)
- [Developing with YUI Test](docs/developer/explanation/javascript/javascript-unittesting.rst)
- [Static assets overview](docs/developer/explanation/static-assets/index.rst)
- [CSS](docs/developer/explanation/static-assets/css.rst)
- [CSS sprites](docs/developer/explanation/static-assets/css-sprites.rst)
- [Favicons](docs/developer/explanation/static-assets/favicon.rst)
- [Images](docs/developer/explanation/static-assets/images.rst)
- [Use MockIo library](docs/developer/how-to/launchpad-development-tips/mockio.rst)
- [CSS style guide](docs/developer/reference/css.rst)

## Services and components

> Deepen your understanding of specific background services and standalone modules supporting Launchpad. This documentation covers the build farm, git/code hosting, mail services, and packaging pipelines.

Reviewer: ```suggestion
Background services and standalone modules that support Launchpad including the build farm, git/code hosting, mail services, and packaging pipelines.
```
